### ISSUE TITLE
Thread safety

### ISSUE THREAD

**Issue by pkamenarsky** [wereHamster/rethinkdb-client-driver#5]
Mainly addressing this: `Note that the driver is neither thread-safe nor reentrant.`

I took a look at the source and the response pool is inside a `TVar`, as well as all other important shared resources. As far as I understand, this should be enough to warrant thread-safety, or am I missing something?

**Comment by wereHamster** [comment]
True, the driver is now probably thread safe. It wasn't in the beginning, before I had the single reader thread and the socket protected by a MVar.

**Comment by wereHamster** [comment]
I would still recommend a resource pool though.
